File: java/src/main/java/io/fivexer/sdk/model/SuggestedWorker.java
package io.fivexer.sdk.model;

import java.util.List;
import java.util.Map;

/**
 * One candidate from a dry-run scoring pass.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class SuggestedWorker {
    private String workerId;
    private boolean eligible;
    private double score;
    private double effectivePriority;
    private List<Map<String, Object>> reasons;  // free-form scoring explanations

    public String getWorkerId() { return workerId; }
    public boolean isEligible() { return eligible; }
    public double getScore() { return score; }
    public double getEffectivePriority() { return effectivePriority; }
    public List<Map<String, Object>> getReasons() { return reasons; }
}

File: php/src/Model/SuggestedWorker.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One candidate from a dry-run scoring pass.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class SuggestedWorker
{
    public function __construct(
        public readonly string $workerId,
        public readonly bool $eligible,
        public readonly float $score,
        public readonly float $effectivePriority,
        /** @var list<array<string, mixed>> */
        public readonly array $reasons,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            eligible: (bool) ($data['eligible'] ?? false),
            score: (float) ($data['score'] ?? 0),
            effectivePriority: (float) ($data['effectivePriority'] ?? 0),
            reasons: $data['reasons'] ?? [],
        );
    }
}
